mousestyles/behavior/__init__.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import print_function, absolute_import, division
+
+
+class BehaviorTree():
+
+    def __init__(self, structure=None, contents={}):
+        self.contents = contents
+        self.structure = structure
+
+    def copy(self):
+        return BehaviorTree(structure=self.structure.copy(),
+                            contents=self.contents.copy())
+
+    def __str__(self):
+        if self.structure is None:
+            return self.contents.__str__()
+        else:
+            return self._display_tree()
+
+    def __getitem__(self, key):
+        return self.contents[key]
+
+    def __setitem__(self, key, value):
+        self.contents[key] = value
+
+    def _display_tree(self, level=0):
+        struct = self.structure
+        if isinstance(struct, list):
+            root = struct[0]
+            children = struct[1]
+        else:
+            root = struct
+            children = []
+
+        result = "\n" + " " * 5 * level + "{}: {}".format(root, self[root])
+
+        for child in children:
+            current_tree = self.copy()
+            current_tree.structure = child
+            result += current_tree._display_tree(level + 1)
+
+        return result
+
+    def summarize(self, f):
+        results = self.copy()
+        for k in results:
+            results[k] = f(results[k])
+        return results
+
+
+def process_raw_intervals(consumption, intervals, active_time,
+                          total_time, epsilon):
+    """
+    Takes `total_consumption` and the data in `intervals` and
+    decomposes it into a tree of behavioral summary statistics.
+
+    Parameters
+    ----------
+    consumption: float
+        total consumption (food, water, or movement) in the time period
+    intervals: Intervals
+        Intervals object representing intervals of behavior
+    epsilon: float
+        tolerance for merging events into bouts
+
+    Returns
+    -------
+    behavior : BehaviorTree
+        dictionary-like object representing breakdown of
+        consumption into various nodes
+
+    Examples
+    --------
+    >>> from mousestyles import data, intervals
+    >>> ints = intervals.Intervals(data.load_intervals('AS'))
+    >>> results = process_raw_intervals(1000, ints, 1)
+    """
+    tree = BehaviorTree(['Consumption',
```

Comment:
I think this should be 'Total Rate' since it's something like 'total drinking / total time' or 'total movement / total time'